examples/taskruns/old-home-volume.yaml

```diff
@@ -0,0 +1,27 @@
+apiVersion: tekton.dev/v1alpha1
+kind: TaskRun
+metadata:
+  generateName: old-home-volume-
+spec:
+  taskSpec:
+    steps:
+    - name: write
+      image: ubuntu
+      command: ['bash']
+      args: ['-c', 'echo some stuff > /builder/home/stuff']
+    - name: read
+      image: ubuntu
+      command: ['cat']
+      args: ['/builder/home/stuff']
+    - name: override-homevol
+      image: ubuntu
+      command: ['bash']
+      # /builder/home/stuff *doesn't* exist, because our specified volumeMount
+      # overrides it.
```

Comment:
im a bit confused by this example - might just take some more comments to explain it tho! I tried to make some guesses about it but then I looked at the code and my guesses were wrong :(

Reply:
This demonstrates that if a step mounts its own volume over `/builder/home` (e.g., to back  it by a PVC), then theirs wins, and the "default" `/builder/home` isn't mounted, just for that step.

Reply:
The way I understand this example, is that while we still support `/builder/home` for backward compatibility, we don't prevent using it as a mount point.
Extra docs would be nice though

Reply:
Added more comments.